File: frontend/src/types/email.ts
export interface EmailSummary {
  id: string;
  from_email: string;
  from_name: string;
  subject: string;
  date: string;
  summary: string;
  priority: 'urgent' | 'normal' | 'fyi';
  category: 'reply_needed' | 'confirm_only' | 'info';
  has_attachment: boolean;
  important_entities: string[];
}

export interface EmailContent {
  id: string;
  from_email: string;
  to_email: string[];
  cc_email: string[];
  subject: string;
  body: string;
  html_body?: string;
  date: string;
  attachments: any[];
}

export interface ReplyDraft {
  to_email: string[];
  cc_email: string[];
  subject: string;
  body: string;
  confidence_score: number;
  reasoning: string;
}

export interface AgentResponse {
  agent_name: string;
  task_type: string;
  success: boolean;
  result: any;
  error_message?: string;
  processing_time: number;
}

export interface WebSocketMessage {
  type: string;
  email_id?: string;
  timestamp: string;
  [key: string]: any;
}

File: backend/app/schemas/email.py
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from datetime import datetime
from enum import Enum

class Priority(str, Enum):
    URGENT = "urgent"
    NORMAL = "normal"
    FYI = "fyi"

class Category(str, Enum):
    REPLY_NEEDED = "reply_needed"
    CONFIRM_ONLY = "confirm_only"
    INFO = "info"

class EmailSummary(BaseModel):
    id: str
    from_email: str
    from_name: str
    subject: str
    date: datetime
    summary: str
    priority: Priority
    category: Category
    has_attachment: bool
    important_entities: List[str] = []

class EmailContent(BaseModel):
    id: str
    from_email: str
    to_email: List[str]
    cc_email: List[str] = []
    subject: str
    body: str
    html_body: Optional[str] = None
    date: datetime
    attachments: List[Dict[str, Any]] = []

class ReplyDraft(BaseModel):
    to_email: List[str]
    cc_email: List[str] = []
    subject: str
    body: str
    confidence_score: float
    reasoning: str

class AgentResponse(BaseModel):
    agent_name: str
    task_type: str
    success: bool
    result: Dict[str, Any]
    error_message: Optional[str] = None
    processing_time: float

class LearningFeedback(BaseModel):
    original_draft: str
    corrected_draft: str
    recipient: str
    correction_type: str
    context: Dict[str, Any]
    timestamp: datetime
